apps/aecore/lib/aecore/channel/channel_off_chain_tx.ex

```diff
@@ -215,13 +219,21 @@ defmodule Aecore.Channel.ChannelOffChainTx do
         encoded_updates,
         state_hash
       ]) do
-    {:ok, channel_id} = Identifier.decode_from_binary(encoded_channel_id)
-    %ChannelOffChainTx{
-      channel_id: channel_id,
-      sequence: :binary.decode_unsigned(sequence),
-      updates: Enum.map(encoded_updates, &ChannelOffChainUpdate.from_list/1),
-      state_hash: state_hash
-    }
+    with {:ok, %Identifier{type: :channel, value: channel_id}} <-
+           Identifier.decode_from_binary(encoded_channel_id) do
```

Comment:
The check for the expected type should be done inside Identifier -> in this way we avoid duplicating the type check code in every tx that de-serializes identifiers.